File: backend/src/main/java/obss/hris/business/abstracts/JobPostService.java
package obss.hris.business.abstracts;

import obss.hris.model.entity.JobPost;
import obss.hris.model.request.CreateJobPostRequest;
import obss.hris.model.request.UpdateJobPostRequest;
import obss.hris.model.response.GetJobPostResponse;

import java.util.List;

public interface JobPostService {

    List<GetJobPostResponse> jobPostsByPage(int page, int size);

    GetJobPostResponse getJobPostByIdForRequest(Long jobPostId);

    JobPost getJobPostById(Long jobPostId);

    List<GetJobPostResponse> getJobPostsByCreatorByPage(String userName, int page, int size);

    JobPost createJobPost(String userName, CreateJobPostRequest jobPost);

    JobPost updateJobPost(UpdateJobPostRequest jobPost);

    void deleteJobPost(Long jobPostId);

    void applyToJobPost(Long jobPostId, Long candidateId);
}

File: frontend/src/services/jobPostService.js
import axios from "axios";
export default class JobPostService {
  async getJobPosts(page, size) {
    return await axios.get(process.env.REACT_APP_JOB_POSTS_URL + `/${page}/${size}`);
  }

  async updateJobPost(token, jobPost) {
    return await axios.put(process.env.REACT_APP_JOB_POSTS_URL, jobPost, {
      headers: {
        Authorization: "Bearer " + token,
      },
    });
  }

  async applyToJobPost(token, jobPostId) {
    return await axios.post(
      process.env.REACT_APP_JOB_POSTS_URL + `/${jobPostId}`,
      {},
      {
        headers: {
          Authorization: "Bearer " + token,
        },
      }
    );
  }

  async deleteJobPost(token, jobPostId) {
    return await axios.delete(process.env.REACT_APP_JOB_POSTS_URL + `/${jobPostId}`, {
      headers: {
        Authorization: "Bearer " + token,
      },
    });
  }
  async createJobPost(token, jobPost) {
    return await axios.post(process.env.REACT_APP_JOB_POSTS_URL, jobPost, {
      headers: {
        Authorization: "Bearer " + token,
      },
    });
  }
}
